### PR TITLE
[PAY-544] Add Transaction Details Modal

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

Adds the Transaction Details Modal

- Depends on #1804 
- Not wired up to BuyAudioModal yet
- Solscan links not wired up yet

image

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

Tested by triggering the actions manually locally

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/web/src/components/transaction-details-modal/components/Block.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import cn from 'classnames'
+
+import styles from './Block.module.css'
+
+export const Block = ({
+  header,
+  children,
+  className,
+  ...divProps
+}: {
```
Comment (dylanjeffers) [inline packages/web/src/components/transaction-details-modal/components/Block.tsx:12]: i do personally like creating the prop type above.

File: packages/web/src/components/transaction-details-modal/components/Block.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+} & ComponentPropsWithoutRef<'div'>) => {
+  return (
+    <div className={cn(styles.block, className)} {...divProps}>
+      <div className={styles.blockHeader}>{header}</div>
+      <div className={styles.blockContent}>{children}</div>
+    </div>
+  )
+}
+
+export const BlockContainer = ({ children }: { children: ReactNode }) => {
```
Comment (dylanjeffers) [inline packages/web/src/components/transaction-details-modal/components/Block.tsx:24]: do we want these to be lists? if so container could be `<ul />` (might want to rename to BlockList), and blocks would render `<li>`

File: packages/web/src/components/transaction-details-modal/components/TransactionPurchaseMetadata.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      <Block
+        header={
+          <>
+            {messages.purchased}
+            <IconButton
+              className={styles.iconButton}
+              icon={<IconExternalLink />}
+              title={messages.viewOnSolScan}
+              aria-label={messages.viewOnSolScan}
+              onClick={handlePurchaseSolScanClicked}
```
Comment (dylanjeffers) [inline packages/web/src/components/transaction-details-modal/components/TransactionPurchaseMetadata.tsx:47]: ideally IconButton renders an `<a>`. The way we can make this work, is have our `Button` component accept an href prop, and if it does, render an `<a>` tag (we do this in native button for reference). and then IconButton use Button, (if it's weirdly overloaded, we can have a ButtonBase clase that handles the `a` tag stuff, and Button and IconButton compose it). That is the ideal setup, faster option is to just make iconbutton handle a tag given href prop. lmk if you need help here!

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-transaction-details-modal

**Review by dylanjeffers — APPROVED** [review]
nice work, some thoughts re lists and anchors

File: packages/web/src/components/transaction-details-modal/components/Block.tsx (hunk 2 above)
Comment (rickyrombo) [reply packages/web/src/components/transaction-details-modal/components/Block.tsx:24]: 🧠

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-transaction-details-modal

**Pull request merged**